services/QuillLMS/app/controllers/integrations_controller.rb

```diff
@@ -0,0 +1,131 @@
+class IntegrationsController < ApplicationController
+  include HTTParty
+  layout "integrations"
+
+  def amplify
+    store_partner_session()
+    @body_class = 'full-width-page white-page'
+    @js_file = 'public'
+    @active_tab = 'Featured Activities'
+    # get list of activities to pass to view
+    @topics = cached_amplify_activities
+  end
+
+  def amplify_all
+    @body_class = 'full-width-page white-page'
+    @js_file = 'public'
+    @active_tab = 'Explore All Activities'
+    @section = if params[:section_id].present? then Section.find(params[:section_id]) else Section.first end
+    @topics = @section.topics.map{ |topic| [topic, topic.activities.production] }.select{ |group| group.second.any? }
+    render :template => 'integrations/amplify'
+  end
+
+  def cached_amplify_activities
+    Rails.cache.fetch(:selected_amplify_activities, :expires_in => 1.day) do
+      format_amplify_activities
+    end
+  end
+
+  def format_amplify_activities
+    activities = selected_amplify_activities
+    topics = activities.map{ |activity| activity.topic }.uniq
+    # The template expects an array of [topic, [activities]] objects, so sort our
+    # activities into the appropriate shape
+    topics.map{ |topic|
+      [topic, activities.map{ |activity|
+        if activity.topic == topic
+          activity
+        end
+      }.compact]
+    }
+  end
+
+  def selected_amplify_activities
+    selected_activities = []
+    selected_amplify_activity_uids.each do |uid|
+      begin
+        activity = Activity.where(uid: uid).first!
```

Comment:
this is just another style thing, but why use a .where and then .first instead of .find_by! ?